### ISSUE TITLE
Hoobs Install on Fedora Failing

### ISSUE THREAD

**Issue by bloody193** [hoobs-org/HOOBS#276]
**Description**
I clean installed hoobs on Fedora Linux on my intel nuc and I get these errors when attempting to launch hoobs.

**Version**
Hoobs 3.0

**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [ ] Yes
- [ x] No

**Pervious version**
If you upgraded, please let us know your previous version.

**Did you orginally upgrade to HOOBS 3 from HOOBS 2.1.1?**
Did you perform an upgrade from HOOBS 2.1.1? Some things are different.
- [ ] Yes
- [x ] No

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [ ] HOOBS Box
- [ ] HOOBS SD Card
- [ x] Custom Install

**If this is a custom install, what's your operating system?**
Please let us know what operating system and version you are using. Ex. Fedora 30 or macOS Catalina.
Fedora 30
![D30856CF-A182-4601-9B4D-4ECEE598FBCD](https://user-images.githubusercontent.com/58402292/70201983-29ee6580-16e6-11ea-8d3e-af12ea2b23b8.jpeg)

**List your plugins**
Please include a list of the plugins you are using. We need to be able to find it on NPM. If the plugin is no published, please include a link to the repository.

**Post your config**
This can be found in the interface, Configuration -> Advanced or you can get it via SSH `cat ~/.hoobs/etc/config.json`

* Remove any passwords from the config file. This is public.

```
POST CONFIG HERE
```

**Post your log**
You can get the log from the intercafe. This is in the Log section.

```
POST LOG HERE
```

**Additional context**

**Comment by mkellsy** [comment]
Did you install Node via DNF or did you install it manually?

DNF installs Node in /usr/bin
Manually installs Node in /usr/local/bin

You may need to install without the install script
https://github.com/hoobs-org/HOOBS/wiki/2.5.3-Installing-Manually

**Comment by bloody193** [comment]
Thanks for the quick reply. I installed it using these instructions. 
https://github.com/hoobs-org/HOOBS/wiki/2.5.1-Fedora-Installation

**Comment by mkellsy** [comment]
Are you still getting the same error?

When you run this `ls -la /usr/bin` do you see node in there?
Then if you run `ls -la /usr/local/bin` do you see node in there?

Also the `hoobs` command should be in `/usr/bin` too. Is it in `/usr/local/bin`.

A side note; I wish Fedora would leave Node alone, it is supposed to be in `/usr/local/bin` but Red Hat has to be different.

PS: I also run Fedora.

**Comment by bloody193** [comment]
I’m getting all sorts of errors now, I’m going to wipe my build and start from scratch. What’s the best way to install it on fedora running on a intel nuc? I

**Comment by mkellsy** [comment]
This is what I do on Fedora 30 and 31 (Server Edition).

The HOOBS install script takes care of Node for you.

Install Bonjour so .local works
```
sudo yum install -y nss-mdns avahi
```

Enable Bonjour
```
sudo systemctl enable avahi-daemon.service
```

Install HOOBS
```
wget -q -O - https://raw.githubusercontent.com/hoobs-org/HOOBS/master/install.sh | sudo bash -
```

**Comment by mkellsy** [comment]
I just ran through it on my vm. Here are some screenshots.

My arch is x86_64
![001](https://user-images.githubusercontent.com/9777975/70346017-8d79af80-181a-11ea-95c6-584d4220e350.png)

Proving Node and NPM are not installed.
![002](https://user-images.githubusercontent.com/9777975/70346034-98344480-181a-11ea-8213-975f0b68c29f.png)

Prereqs
```
sudo dnf install -y nss-mdns avahi
```
![003](https://user-images.githubusercontent.com/9777975/70346076-b39f4f80-181a-11ea-93ce-d98b0205f51f.png)

Enable avahi
```
sudo systemctl enable avahi-daemon.service
```
![004](https://user-images.githubusercontent.com/9777975/70346102-c1ed6b80-181a-11ea-9ecd-c5e19b0f9024.png)

Install HOOBS
```
wget -q -O - https://raw.githubusercontent.com/hoobs-org/HOOBS/master/install.sh | sudo bash -
```
![005](https://user-images.githubusercontent.com/9777975/70346125-d16cb480-181a-11ea-93bd-3750083c05a0.png)

I edit /etc/sudoers so the hoobs user doesn't need a password.
![006](https://user-images.githubusercontent.com/9777975/70346159-e3e6ee00-181a-11ea-9f77-c49cca7d2f83.png)

Results. Note: I have to use the IP because of my VM setup.
![007](https://user-images.githubusercontent.com/9777975/70346188-f2cda080-181a-11ea-8a49-8144f50240ae.png)

I hope this helps.

**Comment by mkellsy** [comment]
Let me know how it goes?

**Comment by bloody193** [comment]
I really appreciate you going above and beyond to help me. I'm going to try this now. I think my problem is due to hoobs not automatically starting up when the system is started, I say this because I discovered if I type "hoobs" in terminal it starts. I edit the sudoers file as noted but maybe i'm doing something wrong. I will attempt from scratch once again. Thanks i'll let you know how it goes.

**Comment by mkellsy** [comment]
Is the `/etc/systemd/system/hoobs.service` there? Is so, you can run `sudo systemctrl enable hoobs`

**Comment by bloody193** [comment]
Still stuck at the same point. I must manually run hoobs. This command " sudo systemctrl enable hoobs" is not found did you mean sudo systemctl enable hoobs.service?

**Comment by mkellsy** [comment]
Yes. I usually tab on that command.

**Comment by bloody193** [comment]
I've followed everything to the T. Everything is installed but when I attempt to log into hoobs is stuck in the splash screen until I manually type in "hoobs" in terminal. Once I do that it loads hoobs.

**Comment by bloody193** [comment]
Heres a error im getting when I check the status.

[Leo@hoobsfedora ~]$ systemctl status hoobs
● hoobs.service - HOOBS
   Loaded: loaded (/etc/systemd/system/hoobs.service; enabled; vendor preset: disabled)
   Active: activating (auto-restart) (Result: exit-code) since Fri 2019-12-06 21:57:02 EST; 1s ago
  Process: 1017 ExecStart=/usr/bin/hoobs (code=exited, status=203/EXEC)
 Main PID: 1017 (code=exited, status=203/EXEC)
      CPU: 2ms
[Leo@hoobsfedora ~]$

**Comment by mkellsy** [comment]
Can run `ls -la /usr/bin` is hoobs in there?
If not `ls -la /usr/local/bin` is hoobs there?

Then edit the hoobs.service file and change `ExecStart=/usr/bin/hoobs` to `ExecStart=/usr/local/bin/hoobs`.

You can also change the User to Leo is you can run it from the terminal.

**Comment by bloody193** [comment]
Once again thanks for your help. I'm re doing fedora from scratch and this time i'm not creating another user. Maybe the problem is due to me installing everything under my name and not the root user ?!

**Comment by bloody193** [comment]
> Can run `ls -la /usr/bin` is hoobs in there?
> If not `ls -la /usr/local/bin` is hoobs there?
> 
> Then edit the hoobs.service file and change `ExecStart=/usr/bin/hoobs` to `ExecStart=/usr/local/bin/hoobs`.
> 
> You can also change the User to Leo is you can run it from the terminal.

This resolved my problem. My install was located in "/usr/local/bin" I edited the hoobs.service file as stated above and now my hoobs starts up on reboot. Thanks Again.

**Comment by mkellsy** [comment]
The init script now detects where NPM installed HOOBS and adjusts the service file.

Fixed in 3.1.2

**Comment by bloody193** [comment]
> The init script now detects where NPM installed HOOBS and adjusts the service file.
> 
> Fixed in 3.1.2

where do i find the "init script" ? I don't find any reading material for version 3.1.2 . I was able to get it to work with the work around you mentioned above by editing the service file but I would like to install it the correct way on my intel nuc.

**Comment by mkellsy** [comment]
It is in /usr/bin. It is called hoobs-init.
